python/samples/getting_started_with_processes/step03/steps/cut_food_with_sharpening_step.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from enum import Enum
+
+from semantic_kernel.functions.kernel_function_decorator import kernel_function
+from semantic_kernel.kernel_pydantic import KernelBaseModel
+from semantic_kernel.processes.kernel_process.kernel_process_event import (
+    KernelProcessEvent,
+)
+from semantic_kernel.processes.kernel_process.kernel_process_step import KernelProcessStep
+from semantic_kernel.processes.kernel_process.kernel_process_step_context import KernelProcessStepContext
+from semantic_kernel.processes.kernel_process.kernel_process_step_state import KernelProcessStepState
+
+
+class CutFoodWithSharpeningState(KernelBaseModel):
```

Comment:
wondering what the approach should be for Step State. So just writing some general thoughts:

- Since `KernelBaseModel` is pydantic based wondering:
    - private props should have a `Field` definition to exclude from serialization if needed?
         - should there be a distinction in python for "public" and "private" props in python?  
    - only allow JSON friendly[int, string, array, obj] types(?) - saw in the `KernelBaseModel` def `arbitrary_types_allowed=True`
    - Any other potential restriction/requirement that may be need for DAPR usage
    - Have another Base class only for processes to enforce certain validations/types (?)

Reply:
These are all good thoughts/questions. In SK in general, we've been following more of the approach of using public attribute/fields, and marking them as `exclude=True` if we don't want them included. We often use the public attribute/field in this way so that Pydantic does perform the model validation on those attributes. We can use `PrivateAttr` on an attribute, but that will bypass Pydantic's model validation. 

For the JSON friendly thought: it's a great point. Right now we do allow complex types, but can potentially run into serialization issues. One thought is that for non-JSON-friendly types, we could leverage json_encoders or define custom methods to ensure proper conversion.

I think we should chat about all of these offline.